### PR TITLE
Add cool night index

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR partially addresses #355 
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [x] `bumpversion (major / minor / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 

This adds the Cool Night Index, based on the formula published in Tonietto & Carbonneau (2004) to xclim.

cni

* **Does this PR introduce a breaking change?** 

No. Just adds an indice and an indicator.

* **Other information**:

Due to the fact that this indicator creates a mosaic of two different months, depending on latitude, one could technically supply a dataset missing all other months and receive the data they demand from the indice. This is not the case for the _indicator_, as the missing data will trigger an invalid year if missing_values checks are enabled. This warning  has been mentioned in the indice docstring.

### CODE REVIEW

File: xclim/data/fr.json [hunk 1]
```diff
@@ -435,6 +434,13 @@
     "comment": "",
     "abstract": "Nombre de jours de vagues de froid. Une vague de froid se produit lorsque la température journalière minimale est sous un seuil durant au moins un certain nombre de jours consécutifs."
   },
+  "COOL_NIGHT_INDEX": {
+    "long_name": "Indice  de  fraîcheur  des  nuits",
+    "description": "Moyenne {freq:f} de la température journalière minimale pendant septembre (hémisphère nord) ou mai (hémisphère sud)",
```
Comment (aulemahal) [inline xclim/data/fr.json:439]:
```suggestion
    "description": "Moyenne {freq:f} de la température journalière minimale en septembre (hémisphère nord) ou en mars (hémisphère sud)",
```

File: xclim/data/fr.json [hunk 2]
```diff
@@ -435,6 +434,13 @@
...
+    "title": "Indice  de  fraîcheur  des  nuits",
+    "comment": "Formule prise de Tonietto & Carbonneau, 2004 (10.1016/j.agrformet.2003.06.001).",
+    "abstract": "indicateur des conditions nycthermiques de maturation, qui correspond à la valeur moyenne de la température minimale de l'air sur les 30 jours précédents à la date de récolte potentielle."
```
Comment (aulemahal) [inline xclim/data/fr.json:442]:
```suggestion
    "abstract": "Indicateur des conditions nycthermiques de maturation, qui correspond à la valeur moyenne de la température minimale de l'air sur les 30 jours précédant la date de récolte potentielle."
```
Cool word "nycthermique"!!!

File: xclim/indicators/atmos/_temperature.py [hunk 3]
```diff
@@ -437,6 +438,18 @@ class TasminTasmax(Daily2D):
     compute=indices.cold_spell_frequency,
 )
 
+cool_night_index = Tasmin(
+    identifier="cool_night_index",
+    units="degC",
+    long_name="cool night index",
+    description="Mean minimum temperature for September (northern hemisphere) or March (southern hemisphere).",
+    cell_methods="time: min within days time: mean over days",
+    comment="Original formula published in Tonietto & Carbonneau, 2004 (10.1016/j.agrformet.2003.06.001).",
```
Comment (aulemahal) [inline xclim/indicators/atmos/_temperature.py:447]:
```suggestion
```
This information is already added in the `reference` attribute!

File: xclim/indicators/atmos/_temperature.py [hunk 4]
```diff
@@ -437,6 +438,18 @@ class TasminTasmax(Daily2D):
...
+    abstract="A night coolness variable which takes into account the mean minimum night temperatures during the "
+    "month when ripening usually occurs beyond the ripening period.",
+    allowed_periods=["A"],
+    compute=wrapped_partial(indices.cool_night_index, suggested=dict(lat=_empty)),
```
Comment (aulemahal) [inline xclim/indicators/atmos/_temperature.py:451]: Is there a reason to manually _force_ `lat` here ?  En fait, the indice is currently broken I think, if you didn't pass `lat`. Even if a lot of use cases would be in only one hemisphere, I think the simplest take on our side would be to make it non-optional directly in the indice.

File: xclim/indices/_agro.py [hunk 5]
```diff
@@ -230,3 +230,53 @@ def biologically_effective_degree_days(
...
+      Mean of daily minimum temperature for month of interest.
+
+    Notes
+    -----
+    Given that this indice only examines September and May months, it possible to send in DataArrays containing only
+    these timesteps. Users should be aware that due to the missing values checks in wrapped Indicators, datasets that
+    are missing several months will be flagged as invalid. This check can be ignored by setting the following context:
+
+    >>> with xclim.set_options(check_missing='skip', data_validation='log'):
+    >>>     cni = xclim.atmos.CNI(...)
```
Comment (aulemahal) [inline xclim/indices/_agro.py:265]:
```suggestion
    >>>     cni = xclim.atmos.cool_night_index(...)  # xdoctest: +SKIP
```

File: xclim/data/fr.json (hunk 2 above)
Comment (Zeitsperre) [reply xclim/data/fr.json]: Pulled it directly from the French version of their article. Indeed, it is pretty "cool".

File: xclim/indicators/atmos/_temperature.py (hunk 3 above)
Comment (Zeitsperre) [reply xclim/indicators/atmos/_temperature.py]: Really? That's convenient. I didn't know!

File: xclim/indicators/atmos/_temperature.py (hunk 4 above)
Comment (Zeitsperre) [reply xclim/indicators/atmos/_temperature.py:450]: Yes, true. It needs to be there (IE: we can't compute without it). I can adjust this.

**Comment by Zeitsperre** [comment]
This is good to go!

File: xclim/data/fr.json [hunk 6]
```diff
@@ -435,6 +434,13 @@
     "comment": "",
     "abstract": "Nombre de jours de vagues de froid. Une vague de froid se produit lorsque la température journalière minimale est sous un seuil durant au moins un certain nombre de jours consécutifs."
   },
+  "COOL_NIGHT_INDEX": {
+    "long_name": "Indice  de  fraîcheur  des  nuits",
+    "description": "Moyenne {freq:f} de la température journalière minimale en septembre (hémisphère nord) ou en mars (hémisphère sud)",
+    "title": "Indice  de  fraîcheur  des  nuits",
```
Comment (aulemahal) [inline xclim/data/fr.json:440]:
```suggestion
    "long_name": "Indice de fraîcheur des nuits",
    "description": "Moyenne {freq:f} de la température journalière minimale en septembre (hémisphère nord) ou en mars (hémisphère sud)",
    "title": "Indice de fraîcheur des nuits",
```
V e r y   s p a c y!

File: xclim/indices/_agro.py [hunk 7]
```diff
@@ -233,6 +237,56 @@ def biologically_effective_degree_days(
...
+      Resampling frequency.
+
+    Returns
+    -------
+    xarray.DataArray, [degC]
+      Mean of daily minimum temperature for month of interest.
+
+    Notes
+    -----
+    Given that this indice only examines September and May months, it possible to send in DataArrays containing only
```
Comment (aulemahal) [inline xclim/indices/_agro.py:265]:
```suggestion
    Given that this indice only examines September and March months, it possible to send in DataArrays containing only
```

**Review by aulemahal — APPROVED** [review]

File: xclim/indices/_agro.py (hunk 7 above)
Comment (Zeitsperre) [reply xclim/indices/_agro.py:265]: Oof. Thanks.

File: xclim/data/fr.json (hunk 6 above)
Comment (Zeitsperre) [reply xclim/data/fr.json:440]: A E S T H E T I C S

**Pull request merged**